### PR TITLE
Jira oauth2

### PR DESCRIPTION
**Pull request by aradnaev**
1. work around expired password-based authentication (asking for API key from user, adding instructions link)
2. fixed email verification link with token
3. work in progress on oauth implementation (jira ouath callbacks)

### CODE REVIEW

File: etabotsite/etabotapp/urls.py [hunk 1]
```diff
@@ -43,7 +50,7 @@
     url(r'^account/password_reset/complete/$',
         auth_views.PasswordResetCompleteView.as_view(),
         name='password_reset_complete'),
-
+    url(r'^jira_callback', jira_callback, name='jira_callback'),
```
Comment (ShanshanHe) [inline etabotsite/etabotapp/urls.py]: can we rename it to something more specific? `jira_callback` seems to generic to me and by reading the code, I don't quite get what it does other than simply return a page saying `OK`.

File: etabotsite/etabotsite/settings.py [hunk 2]
```diff
@@ -220,6 +220,28 @@
 ]
 
 
+# OAuth
+AUTHLIB_OAUTH_CLIENTS = {}
+JIRA_OAUTH = custom_settings.get('JIRA_APP_OAUTH')
```
Comment (ShanshanHe) [inline etabotsite/etabotsite/settings.py]: please update the `custom_settings` in our slack channel as well

File: etabotsite/etabotapp/views.py [hunk 3]
```diff
@@ -186,8 +186,17 @@ def get(self, request, format=None):
             status=status.HTTP_200_OK)
 
 
+def jira_callback(request):
+    """Receieve authorization code from JIRA OAuth."""
+    """Provided as a query parameter called code. This code can be
+    exchanged for an access token"""
+    logging.info(request.GET)
+    # logging.info(request.GET.get('code'))
+    return render(request, 'jira_callback.html')
```
Comment (ShanshanHe) [inline etabotsite/etabotapp/views.py]: I don't see any code that connect with JIRA to get access token. In general, it's good practice to finish a feature before a PR is submitted.

File: etabotsite/etabotapp/urls.py [hunk 4]
```diff
@@ -6,14 +6,21 @@
...
+import logging
+
 
 router = DefaultRouter()
 router.register(r'users', UserViewSet, base_name='users')
 router.register(r'projects', ProjectViewSet, base_name='projects')
 router.register(r'tms', TMSViewSet, base_name='tms')
 
-urlpatterns = [
+urlpatterns = staticfiles_urlpatterns()
```
Comment (ShanshanHe) [inline etabotsite/etabotapp/urls.py]: Would the rearranging having any effect on production deployment?

Comment (aradnaev) [reply etabotsite/etabotapp/urls.py]: generally yes, the order of the urlpatterns matter.
I needed to add this for proper serving of static files.

File: etabotsite/etabotsite/settings.py (hunk 2 above)
Comment (aradnaev) [reply etabotsite/etabotsite/settings.py]: will do. I'll need to figure out prod vs dev auth settings.

File: etabotsite/etabotapp/views.py (hunk 3 above)
Comment (aradnaev) [reply etabotsite/etabotapp/views.py]: this is first phase of development where I use info from request.GET with the code that I exchange manually with curl.
This is a big feature so I am breaking it down into small ones.

File: etabotsite/etabotapp/urls.py (hunk 1 above)
Comment (aradnaev) [reply etabotsite/etabotapp/urls.py]: that is usually how generic callback urls are. The complexity is in the passed parameters. There is one callback per external API.

File: etabotsite/etabotapp/views.py (hunk 3 above)
Comment (aradnaev) [reply etabotsite/etabotapp/views.py]: I'll add TODO comment section so it's clear what's next.

File: etabotsite/etabotsite/settings.py (hunk 2 above)
Comment (aradnaev) [reply etabotsite/etabotsite/settings.py]: created ticket not to forget https://etabot.atlassian.net/browse/ET-577

**Comment by aradnaev** [comment]
@ShanshanHe , thanks for the review comments!

**Pull request merged**